genestack_client/files_util.py

```diff
@@ -748,3 +749,72 @@ def create_dataset(self, name, dataset_type, children, parent=None):
             if parent is None:
                 parent = self.get_special_folder(SpecialFolders.MY_DATASETS)
             return self.invoke('createDataset', parent, name, dataset_type, children)
+
+    def find_metainfo_related_terms(
+            self,
+            file_filter,
+            dictionary_accession,
+            relationship_label,
+            transitive,
+            key,
+            prefix_match=False,
+            search_string='',
+            offset=0,
+            limit=MAX_RELATED_TERMS_LIMIT
+    ):
+        """
+        Find dictionary terms related to metainfo values present on files matching given filter.
+
+        Example usage: find all synonyms of attribute values present on available files.
+        If available files contain values "Human" and "Mouse" for "Organism" attribute,
+        synonyms "Homo sapiens" and "Mus musculus" will be found when searching for "synonym"
+        dictionary relation.
+
+        Another usage is to find all broader (i.e. more general) dictionary terms for attributes
+        present on available files. If available files contain "Brain" and "Lung" for "Tissue"
+        attribute, "Nervous System" and "Respiratory system" will be found when searching for
+        "broader term" dictionary relation.
+
+        Search with `transitive==False` considers only directly related terms, with
+        `transitive==true` transitively-related terms are also found.
+
+        See also MetainfoRelatedValueFilter.
```

Comment:
Use reStructuredText format to reference classes in Python documentation. I would also recommend trying to generate Sphinx documentation to check that docstrings will be formatted correctly in the published documentation (if you aren't doing it already)